File: ui/flutter_ui/lib/models/acp_status.dart
/// Parsed GET /api/v1/acp/agents envelope.
class AcpAgentStatus {
  final String id;
  final bool enabled;
  final bool running;
  final String state;
  final int uptimeS;

  const AcpAgentStatus({
    required this.id,
    required this.enabled,
    required this.running,
    required this.state,
    required this.uptimeS,
  });

  factory AcpAgentStatus.fromJson(Map<String, dynamic> json) {
    return AcpAgentStatus(
      id: json['id'] as String? ?? '',
      enabled: json['enabled'] as bool? ?? false,
      running: json['running'] as bool? ?? false,
      state: json['state'] as String? ?? '',
      uptimeS: json['uptime_s'] as int? ?? 0,
    );
  }
}

class AcpStatusView {
  final bool enabled;
  final List<AcpAgentStatus> agents;

  const AcpStatusView({required this.enabled, required this.agents});

  static const disabled = AcpStatusView(enabled: false, agents: []);

  int get liveCount => agents.where((a) => a.running).length;

  factory AcpStatusView.fromJson(Map<String, dynamic>? json) {
    if (json == null) {
      return disabled;
    }
    final raw = json['agents'];
    final agents = <AcpAgentStatus>[];
    if (raw is List) {
      for (final item in raw) {
        if (item is Map<String, dynamic>) {
          agents.add(AcpAgentStatus.fromJson(item));
        } else if (item is Map) {
          agents.add(AcpAgentStatus.fromJson(Map<String, dynamic>.from(item)));
        }
      }
    }
    return AcpStatusView(
      enabled: json['enabled'] as bool? ?? false,
      agents: agents,
    );
  }
}

File: internal/comm/http/acp_status.go
package http

import (
	"encoding/json"
	"net/http"
	"strings"
)

// acpStatusList is the GET /api/v1/acp/agents envelope.
type acpStatusList struct {
	Enabled bool           `json:"enabled" yaml:"enabled"`
	Agents  []acpStatusRow `json:"agents" yaml:"agents"`
}

type acpStatusRow struct {
	ID      string `json:"id" yaml:"id"`
	Enabled bool   `json:"enabled" yaml:"enabled"`
	Running bool   `json:"running" yaml:"running"`
	State   string `json:"state" yaml:"state"`
	UptimeS int    `json:"uptime_s" yaml:"uptime_s"`
}

func acpDisabledStatus() acpStatusList {
	return acpStatusList{Enabled: false, Agents: []acpStatusRow{}}
}

// handleACPAgentsList handles GET /api/v1/acp/agents.
// Nil rpcCall (or a disabled manager behind acp.list) returns 200 with the
// disabled envelope — never 500.
func (s *Server) handleACPAgentsList(w http.ResponseWriter, r *http.Request) {
	if s.rpcCall == nil {
		s.writeJSON(w, http.StatusOK, acpDisabledStatus())
		return
	}
	result, err := s.rpcCall(r.Context(), "acp.list", json.RawMessage("{}"))
	if err != nil {
		s.writeError(w, http.StatusInternalServerError, strings.ToLower(err.Error()))
		return
	}
	s.writeJSON(w, http.StatusOK, result)
}
